File: soundscape-api/src/main/java/com/sparklecow/soundscape/models/user/UserResponseDto.java
package com.sparklecow.soundscape.models.user;

import lombok.Builder;

import java.time.LocalDate;

import java.util.Set;

@Builder
public record UserResponseDto(
        String username,
        String email,
        String fistName,
        String lastName,
        String profileImageUrl,
        String bannerImageUrl,
        LocalDate birthDate,
        String country,
        Boolean isVerified,
        Boolean isEnabled,
        Boolean isAccountLocked,
        Boolean isCredentialsExpired,
        Set<Role> roles
) {}

File: soundscape-ui/src/app/models/UserResponseDto.ts
import { Role } from "./Role";

export interface UserResponseDto{
  username: string,
  email: string,
  firstName: string,
  lastName: string,
  birthDate: Date,
  country: string,
  profileImageUrl: string,
  bannerImageUrl: string,
  isVerified: boolean,
  isEnabled: boolean,
  isAccountLocked: boolean,
  isCredentialsExpired: boolean,
  role: Role[]
}
